packages/fela-bindings/src/ThemeProviderFactory.js

```diff
@@ -20,9 +20,9 @@ export default function ThemeProviderFactory(
       this.theme = createTheme(props.theme, previousTheme)
     }
 
-    componentWillReceiveProps(nextProps: Object): void {
-      if (!shallowEqual(this.props.theme, nextProps.theme)) {
-        this.theme.update(nextProps.theme)
+    componentDidUpdate(prevProps: Object): void {
```

Comment:
Doesn't this eventually cause an additional update? Before we updated the theme before the actual update happens. Now we update it after the props change => another update.

Reply:
I think this code is equivalent. We update the theme to update all the components that are subscribed to it. Theme update = re-render subscribers. Therefore, there is no difference in the re-rendering of subscribers before the provider has been updated or after.